src/main/java/net/javadiscord/javabot/Bot.java

```diff
@@ -158,7 +198,7 @@ private static void addEventListeners(@NotNull JDA jda, @NotNull DIH4JDA dih4jda
 				new PingableNameListener(),
 				new HugListener()
 		);
-		dih4jda.addListener(new DIH4JDAListener());
+		dih4jda.addEventListener(new DIH4JDAListener());
 	}
 
 	private static void addComponentHandler(@NotNull DIH4JDA dih4jda) {
```

Comment:
You might want to let Spring automatically scan for interaction handlers as well.

Reply:
Well, thats a DIH4JDA and I dont want DIH4JDA to be dependent on Spring

Reply:
You can use Spring for obtaining a list of interaction handlers and listeners and register them automatically.

Reply:
Oh, you're not talking about events, you're talking about the button handlers, select handerls, etc